src/DataProtection/DataProtection/src/AuthenticatedEncryption/CngCbcAuthenticatedEncryptorFactory.cs

```diff
@@ -55,8 +55,9 @@ public CngCbcAuthenticatedEncryptorFactory(ILoggerFactory loggerFactory)
             return null;
         }
 
+        using var key = new Secret(secret);
         return new CbcAuthenticatedEncryptor(
-            keyDerivationKey: new Secret(secret),
+            keyDerivationKey: key,
```

Comment:
I think this is wrong because the key passed in with the returned object will already be disposed when the caller will try to use it.

@halter73 would you agree?

Reply:
Looking at the implementation of `CbcAuthenticatedEncryptor`, it copies the buffer in the constructor and then releases all references to the `Secret`. So, it's probably safe, but it feels like a bit of an implementation detail.

@GrabYourPitchforks right that there is no memory leak, but it does look like it has a `SafeHandle` field that needs to be finalized if you don't dispose it. It does look like this is generally used only once per-key by the singleton KeyRingProvider, so maybe the finalization is okay in order to not rely on the implementation details of the `Encryptor` constructors.

What's your recommendation @GrabYourPitchforks?

Reply:
The general pattern of passing `Secret` instances around is that the receiver should make a standalone copy (via `.ctor(ISecret)`) if they need a copy with a standalone lifetime, which allows the caller to maintain control over their own instance's lifetime. So I don't see any issues with this PR from a "this might break in the future" perspective.

The PR seems to be fine from a correctness / regression perspective.

This PR would prevent the finalization of a few objects: let's ballpark it at single digits count per day. The determination of whether this is valuable I'll leave up to the aspnet team. :)